pkg/server/deploy/build_test.go

```diff
@@ -166,3 +166,25 @@ func TestNewDeploySpec(t *testing.T) {
 		t.Errorf("expected %d, got %d", opts.RevisionHistoryLimit, ds.RevisionHistoryLimit)
 	}
 }
+
+func TestNewDeploySpecDefaultLifecycle(t *testing.T) {
+	expectedDescription := "test"
+	expectedSlugURL := "http://teresa.io/slug.tgz"
+	expectedProcessType := "worker"
+	expectedName := "deploy-test"
+	opts := &Options{RevisionHistoryLimit: 5}
+
+	ds := newDeploySpec(
+		&app.App{Name: expectedName},
+		&TeresaYaml{},
+		st.NewFake(),
+		expectedDescription,
+		expectedSlugURL,
+		expectedProcessType,
+		opts,
+	)
+
+	if ds.Lifecycle == nil {
+		t.Error("expected lifecycle; got nil")
+	}
+}
```

Comment:
What do you think about performing the test against the default value instead of check for a no nil value?

Reply:
Sure, I also will change the test name to reflect better what is being tested,